### PR TITLE
Add retry_file_dump

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

Add a tool to dump `.retry` file.

### CODE REVIEW

**Review by olivielpeau — APPROVED** [review]
thanks!

**Comment by olivielpeau** [comment]
(note: once this is merged let's make sure the tool is documented for Support Engineers)

**Comment by albertvaka** [comment]
Please add Agent Core to the code owners for `tools/retry_file_dump/` since Platform was assigned to this PR.

**Review by albertvaka — APPROVED** [review]

**Pull request merged**
